src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/FunctionCallContent.cs

```diff
@@ -56,6 +56,46 @@ public FunctionCallContent(string callId, string name, IDictionary<string, objec
     [JsonIgnore]
     public Exception? Exception { get; set; }
 
+    /// <summary>
+    /// Creates a new instance of <see cref="FunctionCallContent"/> parsing arguments using a specified encoding and parser.
+    /// </summary>
+    /// <typeparam name="TEncoding">The encoding format from which to parse function call arguments.</typeparam>
+    /// <param name="encodedArguments">The input arguments encoded in <typeparamref name="TEncoding"/>.</param>
+    /// <param name="callId">The function call ID.</param>
+    /// <param name="name">The function name.</param>
+    /// <param name="argumentParser">The parsing implementation converting the encoding to a dictionary of arguments.</param>
+    /// <param name="exceptionFilter">Filters potential parsing exceptions that should be caught and included in the result.</param>
+    /// <returns>A new instance of <see cref="FunctionCallContent"/> containing the parse result.</returns>
+    public static FunctionCallContent CreateFromParsedArguments<TEncoding>(
```

Comment:
This change removes all "helper" code from the Abstractions repo with the exception of this method required by client impls. I used a callback approach to invert dependencies if that's acceptable.

Reply:
Seems ok, though at this point I'm not sure how much benefit this method really provides.